include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1143,27 +1144,84 @@ struct __early_exit_find_or
 // parallel_find_or - sync pattern
 //------------------------------------------------------------------------
 
+template <typename Tag>
+struct __parallel_find_or_n_groups_tuner
+{
+    // Calculate the number of work groups.
+    template <typename _ExecutionPolicy>
+    std::size_t
+    operator()(_ExecutionPolicy&& /*__exec*/, std::size_t __n_groups, const std::size_t /*__wgroup_size*/,
+               const std::size_t /*__rng_n*/) const
+    {
+        return __n_groups;
+    }
+};
+
+// No tuning for FPGA_EMU because we are not going to tune here the performance for FPGA emulation.
+#if !_ONEDPL_FPGA_EMU
+template <>
+struct __parallel_find_or_n_groups_tuner<oneapi::dpl::__internal::__device_backend_tag>
+{
+    // Calculate the number of work groups.
+    template <typename _ExecutionPolicy>
+    std::size_t
+    operator()(_ExecutionPolicy&& /*__exec*/, std::size_t __n_groups, const std::size_t __wgroup_size,
+               const std::size_t __rng_n) const
+    {
+        assert(__rng_n > 0);
+
+        if (__n_groups > 1)
+        {
+            constexpr std::size_t __base_rng_n = 4096;
```

Comment:
I think this is going in a better direction.  

Does it make sense to represent `4096` here as some combination of `max_compute_units`, `[max_]subgroup_size`, or `max_workgroup_size`?

Do you have an understanding of why `sqrt` is appropriate here?

Reply:
We use `sqrt` here to have required (found) best values in some check data size points.